packages/pando-lib/src/scripts.ts

```diff
@@ -0,0 +1,248 @@
+        if (file.remainderPath === '') {
+          modifiedTree[path] = { mode: file.value.mode, blob: file.value.hash }
+        }
+        else {
+          fetchFilesFromTree(file.remainderPath, file.value, modifiedTree)
+        }
+      })
+      resolve(modifiedTree)
+    }
+    catch(err) {
+      reject(err)
+    }
+  })
+}
+
+export interface IAuthorOrCommitter {
+  name: string
+  email: string
+  date: string
+}
+
+export interface ITree { [path: string]: string }
+
+export interface ILinkedDataCommit {
+  author: IAuthorOrCommitter
+  committer: IAuthorOrCommitter
+  message: string
+  parents: CID[] | []
+  tree: CID | IModifiedTree
+  gitType: string
+}
+
+export interface IModifiedTree {
+  [path: string]: { 
+    mode: string | number
+    blob:  CID
+  }
+}
+
+export class Commit {
+  public static async get(cid: CID): Promise<Commit> {
+    return new Promise<Commit>(async (resolve, reject) => {
+      try {
+        let commit = await ipld.get(cid)
+        commit = new Commit(commit)
+        commit['@cid'] = cid
+        commit['@sha'] = cidToSha(cid).toString('hex')
+        resolve(commit)
+      } catch (err) {
+        reject(err)
+      }
+    })
+  }
+
+  public author: IAuthorOrCommitter
+  public committer: IAuthorOrCommitter
+  public message: string
+  public parents: CID[] | []
+  public tree: CID | IModifiedTree
+  public gitType: string
+
+  constructor(commit: ILinkedDataCommit) {
+    this.author = commit.author
+    this.committer = commit.committer
+    this.message = commit.message
+    this.tree = commit.tree
+    this.gitType = commit.gitType
+    this.parents = commit.parents
+  }
+
+  public async put(): Promise<CID> {
+    return new Promise<CID>(async (resolve, reject) => {
+      try {
+        const commitNode = {
+          author: this.author,
+          committer: this.committer,
+          gitType: this.gitType,
+          message: this.message,
+          parents: this.parents,
+          tree: this.tree
+        }
+        const commitBlob = IPLDGit.util.serialize(commitNode)
+        const commitCid  = await IPLDGit.util.cid(commitBlob)
+      
+        const cid = await ipld.put(commitNode, multicodec.GIT_RAW)
+        if (isEqualMultihash(cid, commitCid)) {
+          resolve(cid)
+        }
+        else {
+          throw(Error('CIDs are not the same...'))
+        }
+      } catch (err) {
+        reject(err)
+      }
+    })
+  }
+
+  public async extend(): Promise<void> {
+    return new Promise<void>(async (resolve, reject) => {
+      try {
+        let modifiedTree = {}
+        modifiedTree = await fetchFilesFromTree('', this.tree, modifiedTree)
+        this.tree = modifiedTree
+        resolve()
+      }
+      catch(err) {
+        reject(err)
+      }
+    })
+  }
+}
+
+export class Tree {
+  public static async get(cid: CID): Promise<Tree> {
+    return new Promise<Tree>(async (resolve, reject) => {
+      try {
+        let tree = await ipld.get(cid)
```

Comment:
I would avoid using the same variable for the `IPLD` and `Tree` version cause you can't type them. You can turn it into:
```
const ipldTree: IPLDTree = await ipld.get(cid)
const tree: Tree = new Tree(tree)
```